pkg/daemon/ceph/mds/filesystem.go

```diff
@@ -135,3 +132,97 @@ func DeleteFilesystem(context *clusterd.Context, clusterName, filesystemName str
 	logger.Infof("Removed file system %s", filesystemName)
 	return nil
 }
+
+// PrepareForDaemonUpgrade performs all actions necessary to ensure the filesystem is prepared
+// to have its daemon(s) updated. This helps ensure there is no aberrant behavior during upgrades.
+// If the mds is not prepared within the timeout window, an error will be reported.
+// Ceph docs: http://docs.ceph.com/docs/master/cephfs/upgrading/
+func PrepareForDaemonUpgrade(
+	context *clusterd.Context,
+	clusterName, fsName string,
+	timeout time.Duration,
+) error {
+	logger.Infof("preparing filesystem %s for daemon upgrade", fsName)
+	// * Beginning of noted section 1
+	// This section is necessary for upgrading to Mimic and to/past Luminous 12.2.3.
+	//   See more:  https://ceph.com/releases/v13-2-0-mimic-released/
+	//              http://docs.ceph.com/docs/mimic/cephfs/upgrading/
```

Comment:
Could you change the comment as written in your other PR, keeping the start on one level. Or you could just add another line:
```
// See more:
// http://link1
// http://link2
```